lib/OptimizationODE/src/OptimizationODE.jl

```diff
@@ -0,0 +1,106 @@
+module OptimizationODE
+
+using Reexport
+@reexport using Optimization, SciMLBase
+using DifferentialEquations
+using Optimization.LinearAlgebra
+
+export ODEOptimizer, ODEGradientDescent, RKChebyshevDescent, RKAccelerated, HighOrderDescent
+
+struct ODEOptimizer{T}
+    solver::T
+end
+
+# Solver Constructors (users call these)
+ODEGradientDescent()   = ODEOptimizer(Euler())
+RKChebyshevDescent()   = ODEOptimizer(ROCK2())
+RKAccelerated()        = ODEOptimizer(Tsit5())
+HighOrderDescent()  = ODEOptimizer(Vern7())
+
+
+SciMLBase.requiresbounds(::ODEOptimizer)              = false
+SciMLBase.allowsbounds(::ODEOptimizer)                = false
+SciMLBase.allowscallback(::ODEOptimizer)              = true
+SciMLBase.supports_opt_cache_interface(::ODEOptimizer) = true
+SciMLBase.requiresgradient(::ODEOptimizer)            = true
+SciMLBase.requireshessian(::ODEOptimizer)             = false
+SciMLBase.requiresconsjac(::ODEOptimizer)             = false
+SciMLBase.requiresconshess(::ODEOptimizer)            = false
+
+
+function SciMLBase.__init(prob::OptimizationProblem, opt::ODEOptimizer, data=Optimization.DEFAULT_DATA;
+    dt=nothing, callback=Optimization.DEFAULT_CALLBACK, progress=false,
+    maxiters=nothing, kwargs...)
+
+    return OptimizationCache(prob, opt, data;
+        dt=dt, callback=callback, progress=progress,
+        maxiters=maxiters, kwargs...)
+end
+
+function SciMLBase.__solve(
+  cache::OptimizationCache{F,RC,LB,UB,LC,UC,S,O,D,P,C}
+  ) where {F,RC,LB,UB,LC,UC,S,O<:ODEOptimizer,D,P,C}
+
+    dt    = get(cache.solver_args, :dt, nothing)
+    maxit = get(cache.solver_args, :maxiters, 1000)
+
+    u0 = copy(cache.u0)
+    p  = cache.p
+
+    if cache.f.grad === nothing
+        error("ODEOptimizer requires a gradient. Please provide a function with `grad` defined.")
+    end
+
+    function f!(du, u, p, t)
+        cache.f.grad(du, u, p)
+        @. du = -du
+        return nothing
+    end
+
+    ss_prob = SteadyStateProblem(f!, u0, p)
+
+    algorithm = DynamicSS(cache.opt.solver)
+
+    cb = cache.callback
+    if cb != Optimization.DEFAULT_CALLBACK || get(cache.solver_args,:progress,false) === true
+        function condition(u, t, integrator)
+            true
+        end
+        function affect!(integrator)
+            u_now = integrator.u
+            state = Optimization.OptimizationState(u=u_now, objective=integrator(integrator.t, Val{1}))
```

Comment:
```suggestion
            state = Optimization.OptimizationState(u=u_now, objective=cache.f(integrator.u, integrator.p))
```

Actually, the only `u` in the integrator is the gradient 😅